jormungandr.nix

```diff
@@ -0,0 +1,19 @@
+{ lib, rustPlatform, fetchFromGitHub, pkg-config, openssl, protobuf, rustfmt }:
+let
+  cargoPackage = (builtins.fromTOML (builtins.readFile ./jormungandr/Cargo.toml)).package;
+in
+rustPlatform.buildRustPackage rec {
+  pname = cargoPackage.name;
+  version = cargoPackage.version;
+  src = ./.;
+  cargoSha256 = "sha256-rfgmtKweYxk8BELv2YoF3BsGfdGlQGP8HNLE4xklDHs=";
+  nativeBuildInputs = [ pkg-config protobuf rustfmt ];
```

Comment:
A hint on possible optimization: you can omit `rustfmt` if you build jormungandr without default features
(like `cargo build --manifest-path jormungandr/Cargo.toml --no-default-features` does). `rustfmt` is used by default to format some generated code so that developers can browse it.